scalalib/worker/src/mill/scalalib/worker/ZincWorkerImpl.scala

```diff
@@ -496,6 +504,25 @@ class ZincWorkerImpl(
       .map(path => converter.toVirtualFile(path.toNIO))
       .toArray
 
+    val incOptions = {
+      val platformSpecificFiles: Array[AuxiliaryClassFiles] = platformSuffix match {
+        case s"_sjs$_" => Array(ScalaJSFiles.instance())
+        case s"_native$_" => Array(ScalaNativeFiles.instance())
+        case _ => Array.empty[AuxiliaryClassFiles]
+      }
+
+      val scalaVersionSpecificFiles: Array[AuxiliaryClassFiles] =
```

Comment:
Are auxiliary class files expected to exist? Can't we just always set them for tasty? Is there some API documentation for this?

Reply:
I even think we should not hard-code this setting here, but source it from the actual modules. I'm thinking about semanticdb files for examples, but theoretically, every compiler plugin could provide extra files which should be cleaned up in some way.

Reply:
> I'm curious, can't we just always set them for tasty?

We probably could. I think it was designed this way to avoid unnecessary work when we are certain a specific file can't exist.
Maybe @adpi2 can answer this, since I believe he designed the `AuxiliaryClassFiles` abstraction.

Reply:
> I even think we should not hard-code this setting here, but source it from the actual modules.

This will also change the API. So, to avoid a bunch of new `@deprecated` overloads, we probably want replace the new `platformSuffix` parameter with some to configure the auxiliary files. (These long parameter lists scream for a config class, but we can do that later.)

Reply:
I tried a new approach by adding `protected def auxiliaryClassFileExtensions: T[Seq[String]]` to `ScalaModule` and overriding it in `ScalaJSModule` and `ScalaNativeModule`.

Reply:
Great. I like that there is now less knowledge about specific Scala toolchains in zinc worker. Having it explicitly in the respective modules make them more discoverable, easier to remember when reviewing or refactoring and also more modular,since you don't need to think about Native when implementing JS.

Reply:
About the `protected` keyword: From a bin-compat standpoint, it's not a difference. With `public` targets, we gain the ability to easily inspect a module from cli and from other modules. We're not so much designing a Scala API here but rather a templating a build module. In this context I think encapsulation it less a property, to the contrary. E.g. if we want to transition a build from `ModuleA` to `ModuleB`, it isn't helpful if part of the settings forwarding to the underlying used tools are not readable.

Reply:
Made it public ✅